### ISSUE TITLE
[Critical] Fix target leakage in model training

### ISSUE THREAD

**Issue by kavanaghpatrick** [kavanaghpatrick/london-rental-scraper#94]
### Problem

The V20 model contains **severe target variable leakage** through hardcoded PPSF (price-per-sqft) lookup tables. These values are calculated from the full training set BEFORE train/test splits, causing the model to "see" price information during validation.

### Affected Features (9-10 features)

| Feature | Leakage Type |
|---------|--------------|
| `pc_type_ppsf` | Postcode × type median PPSF from training |
| `log_pc_type_expected_price` | Log(pc_ppsf × sqft) = encoded price |
| `prestige_location_ppsf` | Street-level PPSF (26 locations hardcoded) |
| `prestige_multiplier` | Tier multipliers (2.03×, 1.33×, etc.) |
| `log_prestige_expected_price` | Log(prestige_ppsf × sqft) |
| `prestige_ppsf_ratio` | Ratio to default PPSF |
| `type_ppsf_target` | Overall type PPSF median |
| `log_type_expected_price` | Log(type_ppsf × sqft) |
| `postcode_adjustment` | Derived from training residuals |

### Evidence

**Source of leak** (`rental_price_models_v20.py` lines 126-143):
```python
PC_HOUSE_PPSF = {
    'SW3': 5.46, 'SW7': 6.02, 'SW1X': 8.28, ...
}
```

**How they were calculated** (`add_mews_features_v19.py` lines 400-403):
```python
pc_house_ppsf = df[df['is_house'] == 1].groupby('postcode_district')['ppsf'].median().to_dict()
```

This uses `ppsf = price_pcm / size_sqft` - directly derived from the **target variable**.

### Impact

- Reported metrics (R²=0.908, MAPE=5.4%) are **optimistic**
- True out-of-sample performance likely **5-15% worse**
- Model essentially "memorizes" postcode/street prices rather than learning patterns

### Solution Options

**Option A: Remove leaked features entirely**
- Delete features: `pc_type_ppsf`, `prestige_location_ppsf`, all `*_expected_price` features
- Expected R² drop: 0.05-0.10 but reflects true performance

**Option B: Use proper K-fold target encoding**
```python
from category_encoders import LeaveOneOutEncoder
encoder = LeaveOneOutEncoder(cols=['postcode_district'])
# Encodes using only OTHER samples in each fold
```

**Option C: Replace with external data**
- Use ONS rental index by postcode
- Use tube distance, crime rates, school ratings (not price-derived)
- Use council tax bands as proxy

### Files to Modify

1. `rental_price_models_v20.py` - Remove hardcoded PPSF dictionaries (lines 80-217)
2. `chrome-extension/xgboost.js` - Remove PPSF lookup tables (lines 118-289)
3. `chrome-extension/api/features.json` - Remove leaked feature names

### Acceptance Criteria

- [ ] All PPSF lookup tables removed or replaced with LOO encoding
- [ ] Model retrained without leaked features
- [ ] New metrics reflect true out-of-sample performance
- [ ] Chrome extension updated with clean model

### Priority

**Critical** - Current model metrics are misleading. Must fix before relying on reported accuracy.
